### ISSUE TITLE
Reserve a PURL should register an item

### ISSUE THREAD

**Issue by peetucket** [sul-dlss/happy-heron#464]
The user should be able to reserve a PURL with a new button, which is on the dashboard alongside the "Deposit to this collection" button.  See https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/433047903

When clicked, the action is described here (see p.20 under heading "Secondary button: Reserve a PURL
"): https://docs.google.com/document/d/1fcr2DYo7OrX-qTdeUOTWKSS1rlq_47sehESmbvgUsrk/edit#heading=h.kgkkjfrle9fb 

Basically:
1. Ask user for title (UI to be mocked up by @astridu )
2. Use this title to create a new work
3. Register the object to obtain a druid
4. Save item in database, and then return to dashboard so user can continue to edit

**Comment by jcoyne** [comment]
@astridu should the user be able to set the type of work when they are reserving a PURL?  If they don't, when how should that step happen?

**Comment by astridu** [comment]
They should go to this page to submit that information when they return to edit this deposit (not required for reserving a PURL). https://invis.io/EQXC9CLKCR2#/431445056_v_Deposit_Type_Modal

**Comment by astridu** [comment]
title UI
![image](https://user-images.githubusercontent.com/42216982/105766530-bb362200-5f0e-11eb-80f0-91d3406a51b8.png)
